src/io/flutter/inspector/DiagnosticLevel.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.inspector;
+
+/**
+ * The various priority levels used to filter which diagnostics are shown and
+ * omitted.
+ * <p>
+ * Trees of Flutter diagnostics can be very large so filtering the diagnostics
+ * shown matters. Typically filtering to only show diagnostics with at least
+ * level debug is appropriate.
+ * <p>
+ * See https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/foundation/diagnostics.dart
+ * for the corresponding Dart enum.
+ */
+public enum DiagnosticLevel {
+  /**
+   * Diagnostics that should not be shown.
+   * <p>
+   * If a user chooses to display [hidden] diagnostics, they should not expect
+   * the diagnostics to be formatted consistently with other diagnostics and
+   * they should expect them to sometimes be be misleading. For example,
+   * [FlagProperty] and [ObjectFlagProperty] have uglier formatting when the
```

Comment:
Did we adopt a new javadoc convention? I'm not familiar with [...] in javadoc. I thought we used {@code ...}. If those references should be rewritten then I think it can wait until a later PR.

Reply:
These are copy and pasted from the Dart comments for these properties so the style is a bit wrong. I'll clean it up with a followup PR.